pandas/io/formats/format.py

```diff
@@ -1855,29 +1856,51 @@ def just(x):
     return result
 
 
-def _trim_zeros_complex(
-    str_complexes: np.ndarray, decimal: str = ".", na_rep: str = "NaN"
-) -> List[str]:
+def _trim_zeros_complex(str_complexes: np.ndarray, decimal: str = ".") -> List[str]:
     """
     Separates the real and imaginary parts from the complex number, and
     executes the _trim_zeros_float method on each of those.
     """
     return [
-        "".join(_trim_zeros_float(re.split(r"([j+-])", x), decimal, na_rep))
+        "".join(_trim_zeros_float(re.split(r"([j+-])", x), decimal))
         for x in str_complexes
     ]
 
 
+def _post_process_complex(complex_strings: List[str]) -> List[str]:
+    """
+    Zero pad complex number strings produced by _trim_zeros_complex.
+
+    Input strings are assumed to be of the form " 1.000+1.000j" with a
+    leading space and the real and imaginary parts having equal precision.
+    The output strings will all have length equal to that of the longest
+    input string.
+    """
+    lengths = [len(s) for s in complex_strings]
+    max_length = max(lengths)
+    padded = [
+        s[: -((k - 1) // 2 + 1)]  # real part
+        + (max_length - k) // 2 * "0"
+        + s[-((k - 1) // 2 + 1) : -((k - 1) // 2)]  # + / -
+        + s[-((k - 1) // 2) : -1]  # imaginary part
+        + (max_length - k) // 2 * "0"
+        + s[-1]
+        for s, k in zip(complex_strings, lengths)
+    ]
```

Comment:
Would it be safer to split real and imaginary parts via +- and then process decimal and fractional parts by splitting via the dot? This way you would not need to rely on the symmetry of the original string provided.

Reply:
You mean trim zeros after splitting into fractional non-fractional parts? I think the trimming has to be done with the decimal there. (I realize this helper is very confusing, and there's likely a better way to do this.)

Reply:
Right, I mean trimming zeros after splitting into fractional and non-fractional parts. Since a dot char would always split float number, there is no risk to introduce a bug IMHO (even if there is no dot at all).

Reply:
I think then you have to keep track of which parts are fractional and then only trim those?

However this part is not doing any actual trimming, it's correctly for the fact that the previous function is now trimming "too much" (It trims the real and imaginary parts independently, so they aren't aligned afterwards. Rather than rewrite the other function I found it easier to do this post-processing.)